src/charm.py

```diff
@@ -770,11 +782,23 @@ def tracing_endpoint(self) -> Optional[str]:
             return self.tracing.get_endpoint("otlp_http")
         return None
 
+    @property
+    def logging_endpoints(self) -> Optional[List[str]]:
+        """Loki endpoint for charm logging."""
+        container = self._loki_container
+        if (
+            container.can_connect()
+            and container.get_service(self._name).current is ops.pebble.ServiceStatus.ACTIVE
```

Comment:
- Would it be easier to use [`is_running`](https://github.com/canonical/operator/blob/5582dbbb3713f9e51077d7ab8553adc15c626b0b/ops/pebble.py#L1049)?
- This is an interesting condition. Other charms using charm_logging can't do this check.
What would happen if a loki relation is in place but loki hasn't started/finished refresh yet?

Reply:
- TIL about is_running()!
- it would fail to push logs. Not much we can do about that, same as with all other push models in cos